src/basicfuns.jl

```diff
@@ -191,7 +191,7 @@ function logsumexp(X)
     reduce(logaddexp, X)
 end
 function logsumexp(X::AbstractArray{T}; dims=:) where {T<:Real}
-    u = reduce(max, X, dims=dims, init=log(zero(T)))
+    u = reduce(max, X, dims=dims, init=zero(T))
```

Comment:
This is not right, `X` can be all negative.

Reply:
Thanks for catching the error. Do you know what the correct initial value should be?

Reply:
I think we should not have `init` at all. Just check if `X` is empty and return `log(zero(T))` then. Otherwise, keep going with the `reduce`.

Reply:
Good call. I made the changes you suggested and the results hold for both negative and positive values. Please review if you can.